pkg/kv/bulk/sst_batcher.go

```diff
@@ -585,18 +604,31 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 		}
 	}
 
+	currentBatchStatsCopy := b.currentStats.Identity().(*backuppb.IngestionPerformanceStats)
+	currentBatchStatsCopy.Combine(&b.currentStats)
+	b.currentStats.Reset()
+
 	fn := func(ctx context.Context) error {
 		defer res.Release()
 		defer b.mem.Shrink(ctx, reserved)
-		if err := b.addSSTable(ctx, batchTS, start, end, data, stats, !flushAsync); err != nil {
+		if err := b.addSSTable(ctx, batchTS, start, end, data, stats, !flushAsync, currentBatchStatsCopy); err != nil {
 			return err
 		}
 		b.mu.Lock()
 		summary.DataSize += int64(size)
+		currentBatchStatsCopy.DataSize += int64(size)
 		b.mu.totalRows.Add(summary)
-		atomic.AddInt64(&b.stats.batchWaitAtomic, int64(timeutil.Since(beforeFlush)))
-		atomic.AddInt64(&b.stats.dataSizeAtomic, int64(size))
+
+		afterFlush := timeutil.Now()
+		currentBatchStatsCopy.BatchWait += afterFlush.Sub(beforeFlush)
+		currentBatchStatsCopy.Duration = afterFlush.Sub(b.lastFlush)
+		b.mu.totalStats.Combine(currentBatchStatsCopy)
+		b.lastFlush = afterFlush
```

Comment:
Should we put `lastFlush` in `mu` given that concurrent fns can mutate it?

Reply:
done, put `lastFlush` in `mu`